src/rpc/core_rpc_server.cpp

```diff
@@ -2779,13 +2779,15 @@ namespace cryptonote
       }
     }
 
-    res.height = m_core.get_current_blockchain_height();
-    res.block_hash = string_tools::pod_to_hex(m_core.get_block_id_by_height(res.height - 1));
+    const uint64_t height = m_core.get_current_blockchain_height() - 1;
+
+    res.height = height;
+    res.block_hash = string_tools::pod_to_hex(m_core.get_block_id_by_height(height));
     
     for (auto &pubkey_info : pubkey_info_list)
     {
       COMMAND_RPC_GET_SERVICE_NODES::response::entry entry = {};
-      fill_sn_response_entry(entry, pubkey_info, res.height);
+      fill_sn_response_entry(entry, pubkey_info, height);
```

Comment:
Yeah I had a look over. Since height is used for quorum_cop::calculate_decommission_credit. There are 2 places that this is used. Here in response to an RPC call and in quorum cop in the deregistration logic.

In the deregistration logic the height used is std::max(get_current_blockchain_height(), get_target_blockchain_height()). So this height should be height + 1 and the same below. As long as the height used is consistent it should be fine for reporting back info to the user.

Reply:
Updated.